nominal/core/client.py

```diff
@@ -796,23 +796,32 @@ def create_streaming_connection(
         datasource_id: str,
         connection_name: str,
         datasource_description: str | None = None,
+        nominal_data_source_rid: str | None = None,
         *,
         required_tag_names: list[str] | None = None,
     ) -> Connection:
-        datasource_response = self._clients.storage.create(
-            self._clients.auth_header,
-            storage_datasource_api.CreateNominalDataSourceRequest(
-                id=datasource_id,
-                description=datasource_description,
-            ),
-        )
+        if nominal_data_source_rid is None:
+            datasource_response = self._clients.storage.create(
+                self._clients.auth_header,
+                storage_datasource_api.CreateNominalDataSourceRequest(
+                    id=datasource_id,
+                    description=datasource_description,
+                ),
+            )
+        else:
+            datasource_response = self._clients.storage.batch_get(self._clients.auth_header, [nominal_data_source_rid])[
+                0
+            ]
+            if datasource_response is None:
+                raise ValueError(f"no datasource found with RID {nominal_data_source_rid!r}")
```

Comment:
why would this be None? wouldn't it rather index error if there was no datasource found?